From 0efac9d1e3c3a23a3c5be876817a1f1ea5f42c35 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 03 Jan 2022 08:00:50 +0200
Subject: [PATCH] [SW-68384] gaudi2: security: add debug port access protection

This patch adds support to protect jtag access to all memory
regions (SRAM, HBM, PSOC GLOBAL CONF, ..etc) through debug
access port.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I4b15c389b7421e7832c4bfffdd45db8d1e68785d
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 27ea79d..4bdba2b 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -47,6 +47,7 @@
 #include <jt_mstr_if_core_hbw_regs.h>
 #include <i2c_s_mstr_if_core_hbw_regs.h>
 #include <smi_mstr_if_core_hbw_regs.h>
+#include <psoc_cs_trace_regs.h>
 #include "common.h"
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
@@ -1150,6 +1151,12 @@
 	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
+#define HL_SEC_LOCK_DAP_AP_TO_DBG	BIT(0)
+static void preboot_secure_dbg(void)
+{
+	regmap_writel(HL_SEC_LOCK_DAP_AP_TO_DBG, mmPSOC_CS_TRACE_SECURITY_CTRL);
+}
+
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -1161,6 +1168,7 @@
 		hl_dev_prot_bits_init();
 		preboot_secure_pmmu();
 		preboot_secure_psoc();
+		preboot_secure_dbg();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
